pythonforandroid/recipes/gevent/__init__.py

```diff
@@ -7,4 +8,21 @@ class GeventRecipe(CompiledComponentsPythonRecipe):
     depends = [('python2', 'python3crystax'), 'greenlet']
     patches = ["gevent.patch"]
 
+    def get_recipe_env(self, arch=None, with_flags_in_cc=True):
+        env = super(GeventRecipe, self).get_recipe_env(arch, with_flags_in_cc)
+        # sets linker to use the correct gcc (cross compiler)
+        env['LDSHARED'] = env['CC'] + ' -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions'
```

Comment:
Otherwise the default host linker was picked up rather than the ARM cross compiler.